cartographer/mapping/id_test.cc

```diff
@@ -145,6 +160,32 @@ TEST(IdTest, FindSubmapId) {
   EXPECT_TRUE(map_by_id.find(SubmapId{42, 3}) == map_by_id.end());
 }
 
+TEST(IdTest, LowerBoundEdgeCases) {
+  MapById<SubmapId, Data> map_by_id;
+  map_by_id.Append(0, Data(1));
+  map_by_id.Append(2, Data(2));
+  CHECK(map_by_id.lower_bound(1, CreateTime(10)) ==
+        map_by_id.EndOfTrajectory(1));
+  CHECK(map_by_id.lower_bound(2, CreateTime(3)) ==
+        map_by_id.EndOfTrajectory(2));
+  CHECK(map_by_id.lower_bound(2, CreateTime(1)) ==
+        map_by_id.BeginOfTrajectory(2));
+}
+
+TEST(IdTest, LowerBound) {
+  MapById<SubmapId, Data> map_by_id;
+  map_by_id.Append(0, Data(1));
+  map_by_id.Append(0, Data(2));
+  map_by_id.Append(0, Data(4));
+  map_by_id.Append(0, Data(5));
+  CHECK(map_by_id.lower_bound(0, CreateTime(3)) ==
+        (MapById<SubmapId, Data>::ConstIterator(map_by_id, SubmapId{0, 2})));
+  CHECK(map_by_id.lower_bound(0, CreateTime(2)) ==
+        (MapById<SubmapId, Data>::ConstIterator(map_by_id, SubmapId{0, 1})));
+  CHECK(map_by_id.lower_bound(0, CreateTime(4)) ==
+        (MapById<SubmapId, Data>::ConstIterator(map_by_id, SubmapId{0, 2})));
```

Comment:
CHECK (0, -1000) and (0, 1000).

CHECK negative values for time.
CHECK huge container?

Reply:
Write the simple O(n) solution and test with lots of random data?

Reply:
Done